core/rawdb/accessors_snapshot_ext.go

```diff
@@ -0,0 +1,43 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package rawdb
+
+import (
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/ethdb"
+	"github.com/ava-labs/libevm/log"
+)
+
+// ReadSnapshotBlockHash retrieves the hash of the block whose state is contained in
+// the persisted snapshot.
+func ReadSnapshotBlockHash(db ethdb.KeyValueReader) common.Hash {
+	data, _ := db.Get(snapshotBlockHashKey)
+	if len(data) != common.HashLength {
+		return common.Hash{}
+	}
+	return common.BytesToHash(data)
+}
+
+// WriteSnapshotBlockHash stores the root of the block whose state is contained in
+// the persisted snapshot.
+func WriteSnapshotBlockHash(db ethdb.KeyValueWriter, blockHash common.Hash) {
+	if err := db.Put(snapshotBlockHashKey, blockHash[:]); err != nil {
+		log.Crit("Failed to store snapshot block hash", "err", err)
+	}
+}
+
+// DeleteSnapshotBlockHash deletes the hash of the block whose state is contained in
+// the persisted snapshot. Since snapshots are not immutable, this  method can
+// be used during updates, so a crash or failure will mark the entire snapshot
+// invalid.
+func DeleteSnapshotBlockHash(db ethdb.KeyValueWriter) {
+	if err := db.Delete(snapshotBlockHashKey); err != nil {
+		log.Crit("Failed to remove snapshot block hash", "err", err)
+	}
+}
+
+// IterateAccountSnapshots returns an iterator for walking all of the accounts in the snapshot
+func IterateAccountSnapshots(db ethdb.Iteratee) ethdb.Iterator {
+	return NewKeyLengthIterator(db.NewIterator(SnapshotAccountPrefix, nil), len(SnapshotAccountPrefix)+common.HashLength)
```

Comment:
```suggestion
	return NewKeyLengthIterator(
		db.NewIterator(SnapshotAccountPrefix, nil),
		len(SnapshotAccountPrefix)+common.HashLength,
	)
```

There are a lot of parentheses and it's difficult to parse how they pair up. I found the second argument very confusing.

Reply:
Bundled in 1ccca3b0bdaf5fc3516a95d60596678a3ee78738.
Note these are simply code movements of existing code, but why not change them whilst we're at it indeed 👍